packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -405,9 +405,14 @@ export async function generateBlogPosts(
     await Promise.all(blogSourceFiles.map(doProcessBlogSourceFile))
   ).filter(Boolean) as BlogPost[];
 
-  blogPosts.sort(
-    (a, b) => b.metadata.date.getTime() - a.metadata.date.getTime(),
-  );
+  if (typeof options.sortPosts === 'function') {
+    console.log('options:', options);
+    blogPosts.sort(options.sortPosts);
+  } else {
+    blogPosts.sort(
+      (a, b) => b.metadata.date.getTime() - a.metadata.date.getTime(),
+    );
+  }
 
   if (options.sortPosts === 'ascending') {
     return blogPosts.reverse();
```

Comment:
Can you move all that to a `function sortBlogPosts({blogPosts,sortPosts})` abstraction?

Also refactor: I'd prefer if the `if (options.sortPosts === 'ascending')` case was inside your new `else` branch because it's part of it. That else branch represents all non-function (eg string values, both ascending and descending cases) so it's weird to reverse the order later.

Another possibility could be to have a map "sort presets":

```js
const sortPresets = {
  descending: ({blogPosts}) => {
     blogPosts.sort((a, b) => b.metadata.date.getTime() - a.metadata.date.getTime()); 
  }
  ascending: ({blogPosts}) => {
     blogPosts.sort((a, b) => a.metadata.date.getTime() - b.metadata.date.getTime()); 
  }
}
```

It's possible we'll introduce more presets like that in the future (sort by last update date? see also https://github.com/facebook/docusaurus/issues/8657) so it doesn't look like a bad idea to introduce this now?